core/templates/services/context.service.spec.ts

```diff
@@ -64,6 +64,15 @@ describe('Context service', () => {
       expect(ecs.getQuestionPlayerIsManuallySet()).toEqual(false);
       expect(ecs.isInQuestionPlayerMode()).toEqual(false);
     });
+
+    it('should correctly check the page context', ()=> {
+      expect(ecs.isInExplorationContext()).toBe(true);
```

Comment:
I think this test can be made stronger. isInExplorationContext() outputs true if the context is the exploration player or the exploration editor. Can we change this test such that the context is not an exploration page, then expect that isInExplorationContext() is false, then change it to the player and then expect that it is true? Ditto for exploration editor. Perhaps split it into two it blocks.

Reply:
Note: We also need tests to check that this method returns false if it is not in an exploration page.

Reply:
@kevintab95 
One way is spyOn but its not possible as its already used in beforeEach. 

Another way is to shift the spec down to describe block with testing with other page to return false. 

Or, shifting to describe block testing different pages and use spyOn to change pathname.

What do you feel is the correct approach for this ???

Reply:
Done!